mindmeld/active_learning/data_loading.py

```diff
@@ -0,0 +1,430 @@
+            query_list (ProcessedQueryList): Data structure containing a list of processed queries.
+        Returns:
+            class_labels (List[str]): list of labels for classification task.
+        """
+        if training_level == TRAIN_LEVEL_DOMAIN:
+            return [f"{d}" for d in query_list.domains()]
+        elif training_level == TRAIN_LEVEL_INTENT:
+            return [
+                f"{d}|{i}" for d, i in zip(query_list.domains(), query_list.intents())
+            ]
+        else:
+            raise ValueError(
+                f"Invalid label_type {training_level}. Must be '{TRAIN_LEVEL_DOMAIN}'"
+                f" or '{TRAIN_LEVEL_INTENT}'"
+            )
+
+    @staticmethod
+    def create_label_map(app_path, file_pattern):
+        """Creates a label map.
+
+        Args:
+            app_path (str): Path to MindMeld application
+            file_pattern (str): Regex pattern to match text files. (".*train.*.txt")
+
+        Returns:
+            label_map (LabelMap): A label map.
+        """
+        resource_loader = ResourceLoader.create_resource_loader(app_path)
+        query_tree = resource_loader.get_labeled_queries(label_set=file_pattern)
+        return LabelMap(query_tree)
+
+
+class LogQueriesLoader:
+    def __init__(self, app_path: str, training_level: str, log_file_path: str):
+        """This class loads data as processed queries from a specified log file.
+        Args:
+            app_path (str): Path to the MindMeld application.
+            training_level (str): The hierarchy level to train ("domain" or "intent")
+            log_file_path (str): Path to the log file with log queries.
+        """
+        self.app_path = app_path
+        self.training_level = training_level
+        self.log_file_path = log_file_path
+
+    @staticmethod
+    def filter_raw_text_queries(log_queries_iter) -> List[str]:
+        """Removes duplicates in the text queries.
+
+        Args:
+            log_queries_iter (generator): Log queries generator.
+        Returns:
+            filtered_text_queries (List[str]): a List of filtered text queries.
+        """
+        return list(set(q for q in log_queries_iter))
+
+    def convert_text_queries_to_processed(
+        self, text_queries: List[str]
+    ) -> List[ProcessedQuery]:
+        """Converts text queries to processed queries using an annotator.
+
+        Args:
+            text_queries (List[str]): a List of text queries.
+        Returns:
+            queries (List[ProcessedQuery]): List of processed queries.
+        """
+        logger.info("Loading a Bootstrap Annotator to process log queries.")
+        annotator_params = DEFAULT_AUTO_ANNOTATOR_CONFIG
+        annotator_params["app_path"] = self.app_path
+        bootstrap_annotator = BootstrapAnnotator(**annotator_params)
+        return bootstrap_annotator.text_queries_to_processed_queries(
+            text_queries=text_queries
+        )
+
+    @property
+    def queries(self):
```

Comment:
Why is this function needed when the processedquerylist can take care of it?

Reply:
The queries() function in the LogQueriesLoader loads and converts raw text from the log file to processed_queries. This is different from using the processedquerylist at this stage as the processedquerylist class stored indices of queries that have been seen within the app. To get around this, in the DataBucketFactory, the external log queries are added to the sql database using the get_key() and put() functions in the QueryCache to add new query indices.